subsys/bluetooth/mesh/access.c

```diff
@@ -241,6 +241,15 @@ static void mod_publish(struct k_work *work)
 
 	if (pub->count) {
 		pub->count--;
+
+		if (pub->retr_update && pub->update &&
+		    bt_mesh_model_pub_is_retransmission(pub->mod)) {
+			err = pub->update(pub->mod);
```

Comment:
This looks fine. However, it is not obvious how this `update` callback can be used to introduce changing parameters in retransmissions.  Description of the `update` needs to be updated to tell how retransmission can be detected against the regular periodic publication (and its retransmission itself). Basically by paraphrasing what is shown here.